linalg/superlu.cpp

```diff
@@ -269,10 +269,7 @@ SuperLUSolver::~SuperLUSolver()
       LUstructFree(LUstruct);
    }
 
-   if ( options->SolveInitialized )
-   {
-      dSolveFinalize(options, SOLVEstruct);
-   }
+   dSolveFinalize(options, SOLVEstruct);
```

Comment:
Why remove the check here?

Reply:
This was causing a memory error with address sanitizer, when it checks for options->SolveInitialized.